uint8 AUTO=1
uint8 MANUEL=0
uint8 mode
---
uint8 RESULT_OK=3
uint8 RESULT_ERR=4
uint8 result